### PR TITLE
adding helm configs for supporting postgres

### PR DESCRIPTION
**Pull request by kotharironak**
## Description
Adds helm configs for enabling support for postgres (see more details at hypertrace/hypertrace#124)

### Testing
Generate the helm templates locally running the below the commands 
- helm dependency update ./helm/
- helm template ./helm/

Cases try by changing below values:
- dataStoreType: "postgres"
- dataStoreType: "mongo"

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
This specific change may not require, but when we update in docker-compose for postgres, we will document it.

### CODE REVIEW

File: helm/templates/configmap.yaml [hunk 1]
```diff
@@ -7,12 +7,32 @@ metadata:
...
+        {{- if .Values.configMap.postgres.url }}
+        url={{ .Values.configMap.postgres.url | quote }}
+        {{- else }}
+        host={{ .Values.configMap.postgres.host }}
+        port={{ .Values.configMap.postgres.port }}
+        {{- end }}
+        {{- if .Values.configMap.postgres.database }}
+        database= {{ .Values.configMap.postgres.database }}
+        {{- end }}
+        {{- if .Values.configMap.postgres.user }}
```
Comment (ravisingal) [inline helm/templates/configmap.yaml:30]: we should combine check for both username and password.

Comment (kotharironak) [reply helm/templates/configmap.yaml]: Addressed.

**Review by ravisingal — APPROVED** [review]

File: helm/values.yaml [hunk 2]
```diff
@@ -52,9 +52,14 @@ serviceSelectorLabels:
 
 configMap:
   name: attribute-service-config
+  dataStoreType: "mongo"
```
Comment (avinashkolluru) [inline helm/values.yaml:55]: Can we keep the name of dataStoreType consistent with `mongodb` and `postgres` below.
Either change, the dataStoreType to `mongodb` or change the below config key to `mongo`

File: helm/templates/configmap.yaml [hunk 3]
```diff
@@ -7,12 +7,30 @@ metadata:
...
+      dataStoreType = {{ .Values.configMap.dataStoreType }}
+      {{- if eq .Values.configMap.dataStoreType "mongo" }}
       mongo {
         {{- if .Values.configMap.mongodb.url }}
         url = {{ .Values.configMap.mongodb.url | quote }}
         {{- else }}
         host = {{ .Values.configMap.mongodb.host }}
         {{- end }}
       }
+      {{ else if eq .Values.configMap.dataStoreType "postgres" }}
```
Comment (avinashkolluru) [inline helm/templates/configmap.yaml:19]: Can this be generalized to just pick the key/value pairs under the specific dataStoreType and populate the entire section

Comment (kotharironak) [reply helm/templates/configmap.yaml]: Done

**Review by avinashkolluru — APPROVED** [review]

**Pull request merged**